x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java

```diff
@@ -411,6 +412,31 @@ public void testUserWithNoRolesCanPerformRemoteSearch() throws IOException {
         verifyNoMoreInteractions(auditTrail);
     }
 
+    public void testUserWithNoRolesPerformsRemoteSearchWithScroll() throws IOException {
+        final ParsedScrollId parsedScrollId = mock(ParsedScrollId.class);
+        final boolean hasLocalIndices = randomBoolean();
```

Comment:
I would've made this variable take both values in this test, i.e. don't rely on randomisation.

In principle, it's better to avoid randomisation for test coverage, and use it for variables that don't have any bearing on the output of the test. That's in principle, but sometimes the input value space is large and randomisation might be used on the independent variables as well (more often inside, but sometimes across, value spaces that change the outcome of the test).

Reply:
Thanks that's a good point. I changed it to always test both situations.